nni/algorithms/compression/v2/pytorch/pruning/tools/base.py

```diff
@@ -440,3 +444,185 @@ def _compress_mask(self, mask: Tensor) -> Tensor:
             mask = torch.einsum(ein_expression, mask, torch.ones(self.block_sparse_size).to(mask.device))
 
         return (mask != 0).type_as(mask)
+
+
+@dataclass
+class Task:
+    """
+    Task saves the related information about the task.
+    """
+    task_id: int
+    # the path of the model under pruning in this task
+    model_path: Path
+    # the config_list used in this task
+    config_list: List[Dict]
+    # the path of masks applying on the under pruning model in this task
+    masks_path: Path
+    status: Literal['Pending', 'Running', 'Finished'] = 'Pending'
+    score: Optional[float] = None
+    log_dir: Optional[Path] = None
+    state: dict = field(default_factory=dict)
+
+
+class TaskGenerator:
+    """
+    This class used to generate config list for pruner in each iteration.
+    """
+    def __init__(self, origin_model: Module, origin_config_list: List[Dict] = [],
+                 origin_masks: Dict[str, Dict[str, Tensor]] = {}, log_dir: str = '.', save_result: bool = True):
+        """
+        Parameters
+        ----------
+        origin_model
+            The origin model under pruning.
+        origin_config_list
+            The origin config list provided by the user. Note that this config_list is directly config the origin model.
+            This means the sparsity provided by the origin_masks should also be recorded in the origin_config_list.
+        origin_masks
+            The pre masks apply on the origin model.
+        log_dir
+            The log directory use to saving the task generator log.
+        save_result
+            If saving the intermediate result, including intermediate model and masks during each iteration.
```

Comment:
is there any problem if we make the default value of `save_result` false?

Reply:
no, `save_result` can be set False as default, I will modify it.

Reply:
modified